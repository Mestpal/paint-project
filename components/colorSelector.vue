<template>
  <v-col :cols="4">
    <v-card>
      <v-card-text>
        <v-color-picker
          class="ma-2"
          :swatches="swatcherColors"
          @input="updateSelectedColor($event)"
          show-swatches
        />
      </v-card-text>
    </v-card>
  </v-col>
</template>

<script>
import colorActions from '../mixins/colorActions'

export default {
  mixins: [colorActions]
}
</script>
