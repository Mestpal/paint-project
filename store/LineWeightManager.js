import weightsOptions from '@/config/lines/lines-weights'

export const state = () => ({
  selectedLineWeight: '',
  lineWeightsOptions: weightsOptions.weights
})

export const getters = {
  selectedLineWeight: (state) => { return state.selectedLineWeight },
  lineWeightsOptions: (state) => { return state.lineWeightsOptions }
}

export const mutations = {
  UPDATE_LINE_WEIGHT (state, weight) {
    state.selectedLineWeight = weight
  }
}

export const actions = {
  updateLineSelectedWeight ({ commit }, lineEvent) {
    commit('UPDATE_LINE_WEIGHT', lineEvent.weight)
  }
}
