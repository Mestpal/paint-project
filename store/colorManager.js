import swatcherConf from '@/config/colors/swatcher-colors'

export const state = () => ({
  selectedColor: '',
  swatcherColors: swatcherConf.colors
})

export const getters = {
  selectedColor: (state) => { return state.selectedColor },
  swatcherColors: (state) => { return state.swatcherColors }
}

export const mutations = {
  UPDATE_SELECTED_COLOR (state, colorEvent) {
    state.selectedColor = colorEvent.hex
  }
}

export const actions = {
  updateSelectedColor ({ commit }, colorEvent) {
    commit('UPDATE_SELECTED_COLOR', colorEvent)
  }
}
