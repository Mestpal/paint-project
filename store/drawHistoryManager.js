import { paper } from 'paper'

export const state = () => ({
  drawHistory: [],
  historyStepPointer: 0
})

export const getters = {
  drawHistory: () => { return state.drawHistory }
}

export const mutations = {
  SAVE_STEPS_HISTORY (state, history) {
    state.drawHistory = history
  },
  UPDATE_STEPS_POINTER (state, position) {
    state.historyStepPointer = position
  },
  REDUCE_STEPS_POINTER (state) {
    state.historyStepPointer -= 1
  }
}

export const actions = {
  setCanvasProject () {
    paper.setup(document.getElementById('paint_project_canvan'))
  },
  saveDrawStepsHistory ({ commit }) {
    const history = paper.project.activeLayer.children
    const drawHistory = []

    history.forEach((step, index) => {
      const segmentsHistory = []
      const stepColor = step.strokeColor.components
      const segments = step.segments

      segments.forEach((segment, idx) => {
        segmentsHistory[idx] = [
          segment.point.x,
          segment.point.y
        ]
      })

      drawHistory[index] = {
        segments: segmentsHistory,
        color: stepColor
      }
    })

    const newPointerLocation = drawHistory.length - 1

    commit('SAVE_STEPS_HISTORY', drawHistory)
    commit('UPDATE_STEPS_POINTER', newPointerLocation)
  },
  undoLastDrawStep ({ commit }) {
    const drawStepsLenght = paper.project.activeLayer.children.length
    const lastDrawStepPosition = drawStepsLenght - 1
    if (lastDrawStepPosition >= 0) {
      paper.project.activeLayer.removeChildren(lastDrawStepPosition)
      commit('REDUCE_STEPS_POINTER')
    }
  },
  redoLastDrawStep ({ state, commit }) {
    if (state.drawHistory[state.historyStepPointer + 1]) {
      const redoStep = state.drawHistory[state.historyStepPointer + 1]
      const path = new paper.Path()
      path.strokeColor = new paper.Color(redoStep.color)
      redoStep.segments.forEach((pointData) => {
        const point = new paper.Point(pointData)
        path.add(point)
      })
      commit('UPDATE_STEPS_POINTER', state.historyStepPointer + 1)
    }
  }
}
