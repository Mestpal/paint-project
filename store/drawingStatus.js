export const state = () => ({
  drawStatus: false
})

export const getters = {
  drawStatus: (state) => { return state.drawStatus }
}

export const mutations = {
  CHANGE_DRAW_STATUS (state, drawStatus) {
    state.drawStatus = drawStatus
  }
}

export const actions = {
  updateDrawStatus ({ commit }, status = false) {
    commit('CHANGE_DRAW_STATUS', status)
  }
}
