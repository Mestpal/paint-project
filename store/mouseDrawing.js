export const actions = {
  updateDrawing ({ commit }, data) {
    data.paper.setup(document.getElementById('paint_project_canvan'))
    // Now, draw your things based on component state.
    const point = new data.paper.Point(20, 20)
    const circle = new data.paper.Path.Circle(point, data.diameter / 2)
    circle.fillColor = 'red'
    circle.strokeColor = 'blue'
    console.log(data)
  }
}
