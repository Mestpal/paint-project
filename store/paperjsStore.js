import paper from 'paper'

export const actions = {
  updateDrawing ({ commit }, diameter) {
    paper.setup(document.getElementById('paint_project_canvan'))
    const point = new paper.Point(20, 20)
    const circle = new paper.Path.Circle(point, diameter / 2)
    circle.fillColor = 'red'
    circle.strokeColor = 'blue'
  }
}
