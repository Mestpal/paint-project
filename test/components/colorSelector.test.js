
import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'

import colorSelector from '@/components/colorSelector'
import { getters } from '@/store/colorManager'

Vue.use(Vuetify)
Vue.use(Vuex)
const localVue = createLocalVue()

describe('colorSelector.vue', () => {
  let store
  let state
  let wrapper

  beforeEach(() => {
    state = { swatcherColors: [] }
    store = new Vuex.Store({
      state,
      getters
    })
    wrapper = shallowMount(colorSelector, {
      store,
      localVue
    })
  })

  it('reders the color selector component correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  // it('should emit an event when a color ir selected', () => {
  //   const event = jest.fn()
  //   const color = wrapper.find('v-color-picker-stub')

  //   wrapper.vm.$on('update:color', event)

  //   expect(event).toHaveBeenCalledTimes(0)

  //   // Simulate a click on the color
  //   color.trigger('input')
  //   // wrapper.vm.$nextTick()

  //   // Ensure that our mock event was called
  //   expect(event).toHaveBeenCalledTimes(1)
  // })
})
