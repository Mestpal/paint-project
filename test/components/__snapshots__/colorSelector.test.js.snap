// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colorSelector.vue reders the color selector component correctly 1`] = `
<v-card-stub loaderheight="4" tag="div">
  <v-card-title-stub> Color Tools </v-card-title-stub>
  <v-card-text-stub>
    <v-color-picker-stub canvasheight="150" dotsize="10" mode="rgba" showswatches="true" swatchesmaxheight="150" width="300"></v-color-picker-stub>
  </v-card-text-stub>
</v-card-stub>
`;
